ArduPlane/altitude.cpp

```diff
@@ -295,6 +288,17 @@ void Plane::change_target_altitude(int32_t change_cm)
 void Plane::set_target_altitude_proportion(const Location &loc, float proportion)
 {
     set_target_altitude_location(loc);
+
+    // Only do altitude slope handling when above CLIMB_SLOPE_HGT or when
+    // descending. This is meant to prevent situations where the aircraft tries
+    // to slowly gain height at low altitudes, potentially hitting obstacles.
+    if (target_altitude.offset_cm > 0 &&
+        (adjusted_relative_altitude_cm() <
+         (g2.waypoint_climb_slope_height_min * 100))) {
```

Comment:
i think we want a special case for CLIMB_SLOPE_HGT==0, to skip this code (and in param docs)

Reply:
I guess the argument is that you could set a large negative value to disable rather than having a special case 0 to disable. Since 0 is a valid value. 0 or -1 to disable is a fairly common pattern in AP tho. There is no reason a -2 ect could not still be treated as valid.